src/opm/parser/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -297,30 +459,58 @@ namespace Opm {
         return m_minpvVector;
     }
 
-
     void EclipseGrid::initBinaryGrid(const Deck& deck) {
+
         const DeckKeyword& gdfile_kw = deck.getKeyword("GDFILE");
         const std::string& gdfile_arg = gdfile_kw.getRecord(0).getItem("filename").get<std::string>(0);
         std::string filename = deck.makeDeckPath(gdfile_arg);
 
-        ecl_grid_type * grid_ptr_loc = ecl_grid_load_case__( filename.c_str(), false);
-        if (grid_ptr_loc)
-            this->m_grid.reset( grid_ptr_loc );
-        else
-            throw std::invalid_argument("Failed to load grid from: " + filename);
+        const std::array<int, 3> dims = getNXYZ();
```

Comment:
I have added one new member function in EclipseGrid 

```void initGridFromEGridFile(Opm::EclIO::EclFile& egridfile, std::string fileName);```

and this is called by both the constructor

```EclipseGrid::EclipseGrid(const std::string& fileName )```

and the member function 

```void EclipseGrid::initBinaryGrid(const Deck& deck) ```

I have not used keyword GDFILE at all, but I have done some testing with the commercial simulator this evening and it is actually possible to input COORD, ZCORN, ACTNUM .. keywords both via keywords directly and as part of an Egrid file (via keyword GDFILE). This will not cause the simulator to stop, but a warning is given at standard output. 

We need to discuss further what to do with Flow if a user starts flow with a deck with multiple sources of these grid keywords.